src/form/Calendar/CalendarDays/CalendarDays.tsx

```diff
@@ -134,35 +155,81 @@ export const CalendarDays: FC<CalendarDaysProps> = ({
       // Determine the color variant of the button
       const colorVariant = isSelected ? 'primary' : 'default';
 
-      // Determine the button variant
+      // // Determine the button variant
       const buttonVariant = isSelected ? 'filled' : 'text';
 
+      const isMultiSelectedDateStyle = {
+        ...(isSelected &&
+          isMultiMonths &&
+          !isStartRangeDate &&
+          !isEndRangeDate && {
+            backgroundColor: selectedRangeColor,
+            borderColor: selectedRangerBorderColor
+          })
+      };
+
       return (
         <Button
           key={`day-${ii}`}
           className={classNames(css.day, {
             [css.outside]: day.isNextMonth || day.isPreviousMonth,
             [css.today]: day.isToday,
             [css.range]: isRange && isSelected,
-            [css.startRangeDate]: isRange && isStartRangeDate,
-            [css.endRangeDate]: isRange && isEndRangeDate
+            [isMultiMonths ? css.multiStartRangeDate : css.startRangeDate]:
+              isRange && isStartRangeDate && isMultiDateSelected,
+            [isMultiMonths && css.selectedEndRangeDate]:
+              isRange && isEndRangeDate && isMultiDateSelected,
+            [isMultiMonths ? css.multiEndRangeDate : css.endRangeDate]:
+              isRange && isEndRangeDate
           })}
           onMouseEnter={() => setHoveringDate(day.date)}
           onMouseLeave={() => setHoveringDate(null)}
-          variant={buttonVariant}
-          color={colorVariant}
+          variant={!isMultiMonths ? buttonVariant : 'text'}
+          color={!isMultiMonths ? colorVariant : 'default'}
```

Comment:
not sure about this - the button and color variants shouldn't be set due to multi being set. it feels like you should just correctly set the `buttonVariant` and `colorVariant` props from your app